pkg/golinters/gci.go

```diff
@@ -1,96 +1,43 @@
-	"github.com/shazow/go-diff/difflib"
+	gci "github.com/daixiang0/gci/pkg/analyzer"
 	"golang.org/x/tools/go/analysis"
 
+	"github.com/golangci/golangci-lint/pkg/config"
 	"github.com/golangci/golangci-lint/pkg/golinters/goanalysis"
-	"github.com/golangci/golangci-lint/pkg/lint/linter"
 )
 
 const gciName = "gci"
 
-func NewGci() *goanalysis.Linter {
-	var mu sync.Mutex
-	var resIssues []goanalysis.Issue
-	differ := difflib.New()
+func NewGci(settings *config.GciSettings) *goanalysis.Linter {
+	var linterCfg map[string]map[string]interface{}
 
-	analyzer := &analysis.Analyzer{
-		Name: gciName,
-		Doc:  goanalysis.TheOnlyanalyzerDoc,
-	}
-	return goanalysis.NewLinter(
-		gciName,
-		"Gci control golang package import order and make it always deterministic.",
-		[]*analysis.Analyzer{analyzer},
-		nil,
-	).WithContextSetter(func(lintCtx *linter.Context) {
-		localFlag := lintCtx.Settings().Gci.LocalPrefixes
-		goimportsFlag := lintCtx.Settings().Goimports.LocalPrefixes
-		if localFlag == "" && goimportsFlag != "" {
-			localFlag = goimportsFlag
+	if settings != nil {
+		cfg := map[string]interface{}{
+			gci.NoInlineCommentsFlag:  settings.NoInlineComments,
+			gci.NoPrefixCommentsFlag:  settings.NoPrefixComments,
+			gci.SectionsFlag:          strings.Join(settings.Sections, gci.SectionDelimiter),
+			gci.SectionSeparatorsFlag: strings.Join(settings.SectionSeparator, gci.SectionDelimiter),
 		}
 
-		analyzer.Run = func(pass *analysis.Pass) (interface{}, error) {
-			var fileNames []string
-			for _, f := range pass.Files {
-				pos := pass.Fset.PositionFor(f.Pos(), false)
-				fileNames = append(fileNames, pos.Filename)
-			}
-
-			var issues []goanalysis.Issue
-
-			flagSet := gci.FlagSet{
-				LocalFlag: gci.ParseLocalFlag(localFlag),
-			}
-
-			for _, f := range fileNames {
-				source, result, err := gci.Run(f, &flagSet)
-				if err != nil {
-					return nil, err
-				}
-				if result == nil {
-					continue
-				}
-
-				diff := bytes.Buffer{}
-				_, err = diff.WriteString(fmt.Sprintf("--- %[1]s\n+++ %[1]s\n", f))
-				if err != nil {
-					return nil, fmt.Errorf("can't write diff header: %v", err)
-				}
-
-				err = differ.Diff(&diff, bytes.NewReader(source), bytes.NewReader(result))
-				if err != nil {
-					return nil, fmt.Errorf("can't get gci diff output: %v", err)
-				}
-
-				is, err := extractIssuesFromPatch(diff.String(), lintCtx.Log, lintCtx, gciName)
-				if err != nil {
-					return nil, errors.Wrapf(err, "can't extract issues from gci diff output %q", diff.String())
-				}
-
-				for i := range is {
-					issues = append(issues, goanalysis.NewIssue(&is[i], pass))
-				}
-			}
-
-			if len(issues) == 0 {
-				return nil, nil
-			}
-
-			mu.Lock()
-			resIssues = append(resIssues, issues...)
-			mu.Unlock()
+		if settings.LocalPrefixes != "" {
+			prefix := []string{"standard", "default", fmt.Sprintf("prefix(%s)", settings.LocalPrefixes)}
+			cfg[gci.SectionsFlag] = strings.Join(prefix, gci.SectionDelimiter)
+		}
```

Comment:
We should not overwrite the sections if the new format has been specified
```suggestion
```

Reply:
I think the opposite I don't we to create a mix between old and new configuration (`sections`), I want to force user to only use the new format.

Reply:
"Failing" mean it wouldn't be backwards compatible, it means API change, so new version. nope, better to fall back to the old config.

Reply:
> "Failing" means it wouldn't be backwards compatible, it means API change, so new version. nope, better to fall back to the old config.

It should only fail if old and new format are specified at the same time. Since this is a misconfiguration I think we should handle this case instead of silently using the old config.

Reply:
I added a log to inform about the deprecation, I think it's enough to understand the behavior.